utils/mvtree.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AppResource } from './hostapi';
+
+/**
+ * This interface describes required properties in order to support the quick pick-through-the-tree
+ * logic. It is not required to be implemented, but the quick pick behavior will ignore the
+ * branch of the tree if these properties are not present.
+ */
+export interface QuickPickableModelNode {

```

Comment:
Because it's several related properties, to avoid a chance collision with the implementor's own properties, I'd suggest something like:

```typescript

export interface QuickPickableModelOptions {
    readonly contextValues: string[];
    readonly isLeafNode: boolean;
}

export interface QuickPickableModelNode {
    readonly quickPickOptions: QuickPickableModelOptions;
}

```